pkg/sitewise/api/property_aggregate.go

```diff
@@ -71,13 +91,18 @@ func aggregateQueryToInput(query models.AssetPropertyValueQuery) *iotsitewise.Ba
 	}
 }
 
-func GetAssetPropertyAggregates(ctx context.Context, client client.SitewiseClient, query models.AssetPropertyValueQuery) (*framer.AssetPropertyAggregates, error) {
+func GetAssetPropertyAggregates(ctx context.Context, client client.SitewiseClient, query *models.AssetPropertyValueQuery) (*framer.AssetPropertyAggregates, error) {
```

Comment:
I would suggest returning a modified query, like this:

```golang
func GetAssetPropertyAggregates(ctx context.Context, client client.SitewiseClient,
	query models.AssetPropertyValueQuery) (models.AssetPropertyValueQuery, *framer.AssetPropertyAggregates, error) {

	var (
		maxDps = int(query.MaxDataPoints)
	)

	modifiedQuery, err := getAssetIdAndPropertyId(query, client, ctx)
	if err != nil {
		return models.AssetPropertyValueQuery{}, nil, err
	}

	awsReq := aggregateQueryToInput(modifiedQuery)

	resp, err := client.BatchGetAssetPropertyAggregatesPageAggregation(ctx, awsReq, modifiedQuery.MaxPageAggregations, maxDps)

	if err != nil {
		return models.AssetPropertyValueQuery{}, nil, err
	}

	return modifiedQuery,
		&framer.AssetPropertyAggregates{
			Request: *awsReq,
			Response: iotsitewise.BatchGetAssetPropertyAggregatesOutput{
				SuccessEntries: resp.SuccessEntries,
				SkippedEntries: resp.SkippedEntries,
				ErrorEntries:   resp.ErrorEntries,
				NextToken:      resp.NextToken,
			},
		}, nil
}
```

See suggestion [here](https://github.com/grafana/iot-sitewise-datasource/pull/183/files#diff-af04a694ac71008929ba1d662736c9e3e3460d0f984bb22dbaa02b672c734f67R89-R90)

Reply:
+1 to this also. The function name doesn't imply that query will be changed and there's no real indication that its going to be, so as is it could cause some sneaky issues when debugging.